crates/ra_hir/src/ids.rs

```diff
@@ -171,9 +172,67 @@ impl DefId {
                 Def::Struct(struct_def)
             }
             DefKind::Enum => {
-                let enum_def = Enum::new(self);
+                // FIXME: this isn't great. all this work just to get the `DefId`
```

Comment:
This function is definitely the worst part of this pull request. We have to dive into the syntax tree, get the nodes corresponding to either the `Enum` or the `EnumVariants` and then translate these into `DefId`s. I hope there is a better way of doing this.